contrib/tektoncd/tektoncd-install/base/crds.yaml

```diff
@@ -1,60 +1,86 @@
-apiVersion: apiextensions.k8s.io/v1beta1
+apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   name: clustertasks.tekton.dev
   labels:
-    pipeline.tekton.dev/release: "v0.14.0"
-    version: "v0.14.0"
+    pipeline.tekton.dev/release: "v0.21.0"
 spec:
   group: tekton.dev
   preserveUnknownFields: false
-  validation:
-    openAPIV3Schema:
-      type: object
-      x-kubernetes-preserve-unknown-fields: true
   versions:
-  - name: v1alpha1
-    served: true
-    storage: false
-  - name: v1beta1
-    served: true
-    storage: true
+    - &version
+      name: v1alpha1
+      served: true
+      storage: false
+      schema:
+        openAPIV3Schema:
+          type: object
+          # One can use x-kubernetes-preserve-unknown-fields: true
+          # at the root of the schema (and inside any properties, additionalProperties)
+          # to get the traditional CRD behaviour that nothing is pruned, despite
+          # setting spec.preserveUnknownProperties: false.
+          #
+          # See https://kubernetes.io/blog/2019/06/20/crd-structural-schema/
+          # See issue: https://github.com/knative/serving/issues/912
+          x-kubernetes-preserve-unknown-fields: true
+      # Opt into the status subresource so metadata.generation
+      # starts to increment
+      subresources:
+        status: {}
+    - !!merge <<: *version
```

Comment:
Just an FYI, this notation causes issues with kustomize versions 3.9+, however since the Kubeflow manifests README says to use 3.2.1, this isn't a blocker. I ran into an issue when updating Knative which also has this notation. For reference: https://github.com/kubernetes-sigs/kustomize/issues/3446#issuecomment-797198684